packages/shared/src/stateChainGateway/index.ts

```diff
@@ -0,0 +1,98 @@
+import type { BigNumber, ContractReceipt, Signer } from 'ethers';
+import { StateChainGateway__factory } from '../abis';
+import {
+  getStateChainGatewayContractAddress,
+  getTokenContractAddress,
+  requestApproval,
+} from '../contracts';
+import { ChainflipNetwork } from '../enums';
+import { assert } from '../guards';
+
+type SignerOptions =
+  | { network: ChainflipNetwork; signer: Signer }
+  | {
+      network: 'localnet';
+      signer: Signer;
+      stateChainGatewayContractAddress: string;
+    };
+
+type ExtendLocalnetOptions<T, U> = T extends { network: 'localnet' }
+  ? T & U
+  : T;
+
+export type FundStateChainAccountOptions = ExtendLocalnetOptions<
+  SignerOptions,
+  { flipContractAddress: string }
+>;
+
+export const getStateChainGateway = (options: SignerOptions) => {
+  const stateChainGatewayContractAddress =
+    options.network === 'localnet'
+      ? options.stateChainGatewayContractAddress
+      : getStateChainGatewayContractAddress(options.network);
+
+  return StateChainGateway__factory.connect(
+    stateChainGatewayContractAddress,
+    options.signer,
+  );
+};
+
+export const fundStateChainAccount = async (
+  accountId: `0x${string}`,
+  amount: string,
+  options: FundStateChainAccountOptions,
+): Promise<ContractReceipt> => {
+  const flipContractAddress =
+    options.network === 'localnet'
+      ? options.flipContractAddress
+      : getTokenContractAddress('FLIP', options.network);
+
+  const stateChainGateway = getStateChainGateway(options);
+
+  await requestApproval(
+    flipContractAddress,
+    stateChainGateway.address,
+    amount,
+    options.signer,
+  );
+
+  const transaction = await stateChainGateway.fundStateChainAccount(
+    accountId,
+    amount,
+  );
+
+  const receipt = await transaction.wait(1);
+
+  assert(receipt.status !== 0, 'Transaction failed');
```

Comment:
the `wait` method will already check if the `status` is 0 and throw an error if it is not: https://github.com/ethers-io/ethers.js/blob/f97b92bbb1bde22fcc44100af78d7f31602863ab/packages/providers/src.ts/base-provider.ts#L1548-L1554